ddl/index.go

```diff
@@ -846,48 +891,127 @@ func doReorgWorkForCreateIndex(w *worker, d *ddlCtx, t *meta.Meta, job *model.Jo
 }
 
 func runIngestReorgJob(w *worker, d *ddlCtx, t *meta.Meta, job *model.Job,
-	tbl table.Table, indexInfo *model.IndexInfo) (done bool, ver int64, err error) {
-	bc, ok := ingest.LitBackCtxMgr.Load(job.ID)
-	if ok && bc.Done() {
-		return true, 0, nil
-	}
-	if !ok && job.SnapshotVer != 0 {
-		// The owner is crashed or changed, we need to restart the backfill.
-		job.SnapshotVer = 0
-		job.RowCount = 0
-		return false, ver, nil
-	}
-	bc, err = ingest.LitBackCtxMgr.Register(w.ctx, indexInfo.Unique, job.ID, job.ReorgMeta.SQLMode)
+	tbl table.Table, indexInfo *model.IndexInfo) (
+	done bool, ver int64, err error) {
+	checkpoint, err := loadCheckpoint(w, job)
 	if err != nil {
-		err = tryFallbackToTxnMerge(job, err)
 		return false, ver, errors.Trace(err)
 	}
-	done, ver, err = runReorgJobAndHandleErr(w, d, t, job, tbl, indexInfo, false)
+	if checkpoint == nil {
+		cfg := config.GetGlobalConfig()
+		currentAddr := net.JoinHostPort(cfg.Host, strconv.Itoa(int(cfg.Port)))
+		checkpoint = model.NewReorgCheckpoint([]byte{}, currentAddr)
+	}
+	bc, err := ingest.LitBackCtxMgr.Register(w.ctx, indexInfo.Unique, job.ID, job.ReorgMeta.SQLMode)
```

Comment:
you need also to make some changes local.go:NewLocalBackend,  otherwise if the folder exists the new backend may failed.